packages/service-library/src/servicelib/fastapi/long_running/_client.py

```diff
@@ -0,0 +1,94 @@
+from functools import cached_property
+from typing import Any, Optional
+
+from fastapi import FastAPI, status
+from httpx import AsyncClient, Timeout
+from pydantic import AnyHttpUrl, PositiveFloat
+
+from ._errors import TaskClientResultErrorError
+from ._models import TaskId, TaskResult, TaskStatus
+
+
+class Client:
```

Comment:
**Note 1:**
So I had started with this design. But @sanderegg's reason against it was: we already have a lot of clients, Why not reuse what we have?
That's why the client is not created but provided.

**Note 2:**
You cannot provide the base_url before hand. You still need to be able to overwrite it. Since the base_url changes based on the services it communicates to (dynamic-sidecars always have a different base-url). We can have a default base_url and the option to overwrite it per request. That I can add.

Reply:
I agree with @sanderegg  but one client per service is reasonable. The problem is (as i have seen) when every request has a different client.

In the case of the sidecar, every sidecar is a different service and therefore I would create a different client for each. BUT i will make sure that the lifespan of the client is the same as that of the server.

Reply:
in any case, at the end the idea is to create a composition. If you want to have more flexibility you can also pass the client with the correct ``base_url`` in the constructor.

```python

client = AsyncClient(some_service_base_url)
# using client for other entrypoints
await client.get("/health")

# using client for a long running task
client_long_a = long_running_client.Client(client, prefix="/long-a")
client_long_b = long_running_client.Client(client, prefix="/long-b")

``

Reply:
Regarding having 1 client per server. That's one of the planned changes we had in mind, but it is out of the scope of this PR. I'll address it when integrating this PR in the director-v2 and dy-sidecar.

Also please note that the `Client` is initialised upon setup. So the clients will still have to passed down upon each request.

Reply:
> in any case, at the end the idea is to create a composition. If you want to have more flexibility you can also pass the client with the correct `base_url` in the constructor.
> 
> ```python
> client = AsyncClient(some_service_base_url)
> # using client for other entrypoints
> await client.get("/health")
> 
> # using client for a long running task
> client_long = long_running_client.Client(client)
> 
> task_id = await client.get("/long-a").json()
> r = await client_long. get_task_status("/long-a", task_id)
> 
> task_id = await client.get("/long-b").json()
> r = await client_long. get_task_status("/long-b", task_id)
> ```

Your idea is nice. I think I can decouple the client from the lifetime of the application. I like this pattern of composing.